**Автор: Анна Смелова**

## Решение домашнего задания к уроку 2 “Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона”

In [1]:
from math import factorial, e

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Задача
#### Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

**<font color='darkblue'>Решение:</font>**<br>
Случайная величина - попадание в мишень: попал/не попал, испытаний сравнительно мало, вероятность попадания высокая $\Rightarrow$ формула Бернулли с $p=0.8$, $n=100$ и $k=85$<br>
$P(X=k)=C_n^kp^k(1-p)^{n-k}$<br><br>
$P(X=85)=C_{100}^{85}\cdot0.8^{85}\cdot0.2^{15}=\frac{100!\cdot0.8^{85}\cdot0.2^{15}}{85!\cdot15!}$

In [3]:
combinations(100, 85) * 0.8**85 * 0.2**15

0.048061793700746556

$P(X=85)=0.048061793700746556$

**<font color='darkblue'>Ответ: вероятность того, что стрелок попадет в цель ровно 85 раз $0.048061793700746556$.</font>**

### 2. Задача
#### Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

**<font color='darkblue'>Решение:</font>**<br>
Случайная величина - перегорание лампочки: горит/не горит, испытаний много, вероятность события низкая $\Rightarrow$ формула Пуассона с $n=5000$, $p=0.0004$, $m_1=0$ и $m_2=2$ <br>
$P_m\approx\frac{\lambda^m}{m!}e^{-\lambda}$, где $\lambda=n\cdot p$<br><br>
$\lambda=0.0004\cdot 5000=2$<br>
$P_{m_1}=P_0=\frac{2^0}{0!}e^{-2}=e^{-2}=\frac{1}{e^2}$<br>
$P_{m_2}=P_2=\frac{2^2}{2!}e^{-2}=2e^{-2}=\frac{2}{e^2}$

In [4]:
1 / e**2

0.1353352832366127

In [5]:
2 / e**2 

0.2706705664732254

**<font color='darkblue'>Ответ: вероятность, что ни одна из лампочек не перегорит в первый день $0.1353352832366127$, вероятность, что перегорят ровно две $0.2706705664732254$</font>**

### 3. Задача
#### Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

**<font color='darkblue'>Решение:</font>**<br>
Формула Бернулли с $n=144$, $k=70$ и $p=0.5$:<br>
$P(X=k)=C_n^kp^k(1-p)^{n-k}$<br><br>
$P(X=70)=C_{144}^{70}\cdot0.5^{70}\cdot0.5^{74}=\frac{144!\cdot0.5^{144}}{70!\cdot74!}$

In [6]:
combinations(144, 70) * 0.5**144

0.06281178035144776

$P(X=70)=0.06281178035144776$

**<font color='darkblue'>Ответ: вероятность того, что орел выпадет ровно 70 раз $0.06281178035144776$.</font>**

### 4. Задача
#### В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

**<font color='darkblue'>Решение:</font>**<br>
1. Какова вероятность того, что все мячи белые?<br>
Вероятность вытащить 2 белых мяча из первого ящика: $P(2whitefrom1)=\frac{7}{10}\cdot\frac{6}{9}=\frac{42}{90}=\frac{14}{30}=\frac{7}{15}$<br>
Вероятность вытащить 2 белых мяча из второго ящика: $P(2whitefrom2)=\frac{9}{11}\cdot\frac{8}{10}=\frac{72}{110}=\frac{36}{55}$<br>
Вероятность вытащить все белые мячи: $P(allwhite)=P(2whitefrom1)\cdot P(2whitefrom2)=\frac{7\cdot36}{15\cdot55}$

In [7]:
(7 * 36) / (15 * 55)

0.3054545454545455

$P(allwhite)=0.3054545454545455$<br><br>
2. Какова вероятность того, что ровно два мяча белые?<br>
Ситуация 1: два белых из 1 ящика и два черных из 2 ящика: $P(2-0)=\frac{7}{10}\cdot\frac{6}{9}\cdot\frac{2}{11}\cdot\frac{1}{10}$<br>
Ситуация 2: два черных из 1 ящика и два белых из 2 ящика: $P(0-2)=\frac{3}{10}\cdot\frac{2}{9}\cdot\frac{9}{11}\cdot\frac{8}{10}$<br>
Ситуация 3: один белый из 1 ящика и один белый из 2 ящика: $P(1-1)=\frac{7}{10}\cdot\frac{3}{9}\cdot\frac{9}{11}\cdot\frac{2}{10}$<br>
Вероятность вытащить ровно два белых мяча: $P(=2white)=P(2-0)+P(0-2)+P(1-1)$

In [8]:
(7 * 6 * 2) / (10 * 9 * 11 * 10) + (3 * 2 * 9 * 8) / (10 * 9 * 11 * 10) + (7 * 3 * 9 * 2) / (10 * 9 * 11 * 10)

0.0903030303030303

$P(=2white)=0.0903030303030303$<br><br>
3. Какова вероятность того, что хотя бы один мяч белый?<br>
Рассмотрим обратный случай - все мячи черные:<br>
Вероятность вытащить 2 черных мяча из первого ящика: $P(2blackfrom1)=\frac{3}{10}\cdot\frac{2}{9}=\frac{6}{90}=\frac{1}{15}$<br>
Вероятность вытащить 2 черных мяча из второго ящика: $P(2blackfrom2)=\frac{2}{11}\cdot\frac{1}{10}=\frac{2}{110}=\frac{1}{55}$<br>
Вероятность вытащить все черные мячи: $P(allblack)=P(2blackfrom1)\cdot P(2blackfrom2)=\frac{1\cdot1}{15\cdot55}$<br>
Вероятность того, что хотя бы один мяч белый: $P(\geq1white)=1-P(allblack)=1-\frac{1}{15\cdot55}$

In [9]:
1 - 1 / (15 * 55)

0.9987878787878788

$P(\geq1white)=0.9987878787878788$

**<font color='darkblue'>Ответ: вероятность того, что все мячи белые $0.3054545454545455$, вероятность того, что ровно два мяча белые $0.0903030303030303$, вероятность того, что хотя бы один мяч белый $0.9987878787878788$.</font>**